man/notify-selfcontained-example.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: MIT-0
+#
+# Implement the systemd notify protocol without external dependencies.
+# Supports both readiness notification on startup and on reloading,
+# according to the protocol defined at:
+# https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html
+# This protocol is guaranteed to be stable as per:
+# https://systemd.io/PORTABILITY_AND_STABILITY/
+
+import errno
+import os
+import signal
+import socket
+import time
+
+reloading = False
+terminating = False
+
+
+def notify(message):
+    if message is None or len(message) == 0:
+        raise OSError(errno.EINVAL, "notify() requires a message")
```

Comment:
```python
if not message:
    raise ValueError('notify() requires a message')
```

Reply:
I generally prefer being explicit about testing for invalid values, but in this context it makes sense to be more concise.